src/play-item.js

```diff
@@ -32,14 +32,20 @@ const clearTracks = (player) => {
 const playItem = (player, item) => {
   const replay = !player.paused() || player.ended();
 
-  player.trigger('beforeplaylistitem', item);
+  player.trigger('beforeplaylistitem', item.originalValue || item);
+
+  if (item.playlistItemId_) {
+    player.playlist.currentPlaylistItemId_ = item.playlistItemId_;
+  }
```

Comment:
This needs to be set before the `player.ready` and definitely before the `playlistitem` event (moved out from `playlist-maker.js`).